yt_dlp/extractor/kanopy.py

```diff
@@ -0,0 +1,127 @@
+import json
+
+from .common import InfoExtractor
+from ..utils import ExtractorError, traverse_obj, try_call, url_or_none
+
+
+class KanopyIE(InfoExtractor):
+    _API_BASE_URL = 'https://www.kanopy.com/kapi/'
+    _NETRC_MACHINE = 'kanopy'
+    _VALID_URL = r'https?://(?:www\.)?kanopy.com/(?P<lang>\w{2})/(?P<institution>.*?)/watch/video/(?P<id>[0-9]+)'
+    _TESTS = [
+        {
+            'url': 'https://www.kanopy.com/en/pleasantvalley/watch/video/114603',
+            'info_dict': {
+                'id': '114603',
+                'ext': 'mp4',
+                'title': 'The Blue Kite',
+                'description': 'md5:6163af52ae92627ae7c58906991d3792',
+            },
+        },
+    ]
+    _LOGIN_REQUIRED = True
+    _ACCESS_TOKEN = None
+    _USER_ID = None
+
+    headers = {
+        'content-type': 'application/json',
+        'x-version': 'web/prod/4.3.0/2024-01-08-15-13-05',
+    }
+
+    def _match_membership(self, memberships, matcher):
+        return next((d for d in memberships if d['subdomain'] == matcher), None)
+
+    def _real_initialize(self):
+        if not self._ACCESS_TOKEN and not self._USER_ID:
+            self.raise_login_required()
+
+    def _perform_login(self, username, password):
+        self.report_login()
+        try:
+            access_json = self._download_json(
+                self._API_BASE_URL + 'login',
+                None,
+                'Logging in to site using credentials',
+                'Unable to log in',
+                fatal=False,
+                headers=self.headers,
+                data=json.dumps(
+                    {
+                        'credentialType': 'email',
+                        'emailUser': {'email': username, 'password': password},
+                    }
+                ).encode(),
+            )
+            self._ACCESS_TOKEN = try_call(lambda: access_json['jwt'])
+            if self._ACCESS_TOKEN is None:
+                self.report_warning('Failed to get Access token')
+            else:
+                self.headers.update({'Authorization': 'Bearer %s' % self._ACCESS_TOKEN})
+                self._USER_ID = try_call(lambda: access_json['userId'])
+        except ExtractorError as e:
+            self.report_warning(e)
```

Comment:
This should be unnecessary. Above, the api call has `fatal=False` and the other steps should never throw. However, making this non fatal is an odd choice. Is the login common to fail? If not, then throwing an actual error is more appropriate